validation/src/main/scala/hmda/validation/rules/ts/validity/V112.scala

```diff
@@ -0,0 +1,21 @@
+package hmda.validation.rules.ts.validity
+
+import hmda.model.fi.ts.TransmittalSheet
+import hmda.validation.dsl.Result
+import hmda.validation.rules.EditCheck
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+
+/*
+ Parent zip code must be in the proper format
+ */
+object V112 extends EditCheck[TransmittalSheet] {
+
+  override def apply(ts: TransmittalSheet): Result = {
+    val regex = "^(?:\\d{5}(?:-\\d{4})?)?$".r // Matches "", "12345", or "12345-6789"
```

Comment:
I'm only taking a super quick look, but I agree this should probably go into `PredicateRegex`